src/app/(nextjs_migration)/components/client/PageLoad.tsx

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { useEffect } from "react";
+import { useGlean } from "../../../hooks/useGlean";
+
+interface PageLoadProps {
+  pageTitle: string;
+}
+
+// Empty component that records a page view on first load.
+export const PageLoad = ({ pageTitle }: PageLoadProps) => {
+  const { appEvents } = useGlean();
+
+  // On first load of the page, record a page view.
+  useEffect(() => {
+    if (pageTitle) {
+      appEvents.pageView.record({ label: pageTitle });
+    }
+  }, [appEvents, pageTitle]);
+
+  // This component doesn't render anything.
+  return <></>;
```

Comment:
This makes me think it probably makes more sense to make this a hook, rather than a component?

Reply:
I made this a component rather than a hook since I am using this on pages like `src/app/(nextjs_migration)/(guest)/page.tsx`. If this was a react component, then yes we could use it as a hook.

I am not an expert with SSR projects, is there a way that we could use a hook on that page?

Reply:
Ah gotcha. Well, I'd caveat this with that everyone's new to the whole server/client component shift, but... Yeah, it'd need a client component I think. There aren't yet estasblished patterns and best practices, but I think that I'd still prefer the functionality in a hook, and then having this component only load that hook - to clarify that the only reason the component is there, is to make the hook run client-side. But not a blocker :)